rm-main/src/ui/tabs/search.rs

```diff
@@ -50,21 +54,67 @@ impl SearchTab {
         let ctx_clone = ctx.clone();
         let table_clone = Arc::clone(&table);
         let search_result_info_clone = Arc::clone(&search_result_info);
+        let state = Arc::new(Mutex::new(throbber_widgets_tui::ThrobberState::default()));
+
         tokio::task::spawn(async move {
-            let magnetease = Magnetease::new();
             while let Some(search_phrase) = rx.recv().await {
-                search_result_info_clone.lock().unwrap().searching();
+                search_result_info_clone
+                    .lock()
+                    .unwrap()
+                    .searching(Arc::clone(&state));
                 ctx_clone.send_action(Action::Render);
-                let res = magnetease.search(&search_phrase).await;
-                if res.is_empty() {
-                    search_result_info_clone.lock().unwrap().not_found();
-                } else {
-                    search_result_info_clone.lock().unwrap().found(res.len());
+
+                let state_clone = Arc::clone(&state);
+                let mut set = JoinSet::new();
+                set.spawn(async move {
+                    let magnetease = Magnetease::new();
+                    let res = magnetease.search(&search_phrase).await;
+                    res
+                });
+
+                set.spawn(async move {
+                    let tick_rate = tokio::time::Duration::from_millis(100);
+                    let mut last_tick = tokio::time::Instant::now();
+                    loop {
+                        if last_tick.elapsed() >= tick_rate {
+                            state_clone.lock().unwrap().calc_next();
+                            last_tick = tokio::time::Instant::now();
+                            // ctx_clone_next.send_action(Action::Render);
+                        }
+                    }
+                });
+
+                while let Some(res) = set.join_next().await {
+                    match res {
+                        Ok(res) => {
+                            if res.is_empty() {
+                                search_result_info_clone.lock().unwrap().not_found();
+                            } else {
+                                search_result_info_clone.lock().unwrap().found(res.len());
+                            }
+
+                            // TODO: add an X icon if no results, else V when results
+                            table_clone.lock().unwrap().set_items(res);
+                            ctx_clone.send_action(Action::Render);
+                        }
+                        Err(e) => {}
+                    }
                 }
```

Comment:
@micielski im trying to create a thread that updates the position of the loading spinner every 100ms - but the main thread seems to get stuck in the loop? is there a better way of doing this?

Reply:
Defining a tick() method in the component trait, and caling it from the app main loop every 100ms or so. But you still have to find a way to only draw this one widget instead of the whole window to make it reasonably efficient

Reply:
it should be okay to re-render the entire window every 100-200ms no? apparently ratatui only renders diffs so in the case where only the spinner's state changes it shouldn't be an issue?

Reply:
It doesn't matter whether ratatui only renders diffs; there's still widgets creation (especially table), mutexes etc

Reply:
according to https://github.com/ratatui-org/ratatui/discussions/409, it seems like only rendering specific widgets isn't possible in ratatui. 

i can experiment with rendering the entire window every 100-200ms when the loading indicator is rendered and see how it goes first

Reply:
The more items in a table the worse it gets:  https://github.com/ratatui-org/ratatui/issues/1004
I'm currently in the process of planning on how to refactor this app to make state and ui more separate.

Reply:
PR currently adds the `Tick` action that triggers a re-render if the current `SearchResultState` is `SearchResultState::Searching` 

could u help me test if there are any performance issues on your end ? would be good to know the amount of torrents (items in the list) that begin causing performance/rendering issues

Reply:
I'll investigate the performance later with more torrents because FreeBSD for some reason reports me much smaller CPU usage than Linux